Deepak_Yadav/tic tac toe/scripts.js

```diff
@@ -0,0 +1,170 @@
+    
+
+    if(((b1=="X") || (b1=="O")) && ((b1 == b2) && (b2 == b3))){
+        highLight("b1", "b2", "b3");
+    }
+    else if(((b1=="X") || (b1=="O")) && ((b1 == b4) && (b4 == b7))){
+        highLight("b1", "b4", "b7");
+    }
+    else if(((b1=="X") || (b1=="O")) && ((b1 == b5) && (b5 == b9))){
+        highLight("b1", "b5", "b9");
+    }
+    else if(((b2=="X") || (b2=="O")) && ((b2 == b5) && (b5 == b8))){
+        highLight("b2", "b5", "b8");
+    }
+    else if(((b3=="X") || (b3=="O")) && ((b3 == b6) && (b6 == b9))){
+        highLight("b3", "b6", "b9");
+    }
+    else if(((b3=="X") || (b3=="O")) && ((b3 == b5) && (b5 == b7))){
+        highLight("b3", "b5", "b7");
+    }
+    else if(((b4=="X") || (b4=="O")) && ((b4 == b5) && (b5 == b6))){
+        highLight("b4", "b5", "b6");
+    }
+    else if(((b7=="X") || (b7=="O")) && ((b7 == b8) && (b8 == b9))){
+        highLight("b7", "b8", "b9");
+    }
+}
+
+
+//Disables the buttons, and paints the button's ORANGE, which are in WINNING pattern.
+//It takes buttons id's as the three arguments and symbol=X or O as another argument, but the symbol īs currently not in use.
+
+//The button pressed will be disabled and symbol 'X' or 'O' will appear on it
+function setButton(button, options){
+    if(button == 1){
+        var btn = document.getElementById("b1");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    else if(button == 2){
+        var btn = document.getElementById("b2");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    else if(button == 3){
+        var btn = document.getElementById("b3");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    else if(button == 4){
+        var btn = document.getElementById("b4");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    else if(button == 5){
+        var btn = document.getElementById("b5");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    else if(button == 6){
+        var btn = document.getElementById("b6");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    else if(button == 7){
+        var btn = document.getElementById("b7");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    else if(button == 8){
+        var btn = document.getElementById("b8");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    else if(button == 9){
+        var btn = document.getElementById("b9");
+        btn.innerHTML = options;
+        btn.disabled = true;
+    }
+    //Checks if a WINNING pattern is made
+    checker();
+}
+
+//for alternate inputs 
+function clicker(button){
+    if(options == "X"){
+        options = "O";
+        setButton(button, options);
+    }
+    else if(options == "O"){
+        options = "X";
+        setButton(button, options);
+    }
+}
+
+function btnDisable(){
+    document.getElementById("b1").disabled = true;
```

Comment:
missing DRY

Reply:
will apply this before eod.
thanks nikhil for the feedback.